enhancements/network/user-defined-network-segmentation.md

```diff
@@ -513,21 +512,33 @@ type UserDefinedNetworkSpec struct {
     // When omitted, this means no opinion and the platform is left to choose a reasonable default which is subject to change over time.    
     // +kubebuilder:validation:XValidation:rule="1 <= size(self) && size(self) <= 2", message="Unexpected number of join subnets"
     // +optional
-    JoinSubnets []string `json:"joinSubnets,omitempty"`
+    JoinSubnets []string `json:"joinSubnets,omitempty"`qq
     
     // Control IP addresses management lifecycle.
     // When `Persistent` is specified it enable workloads have persistent IP addresses.
     // For example: Virtual Machines will have the same IP addresses along their lifecycle (stop, start migration, reboots).
     // Supported by Topology `Layer2` and `Localnet`.
     // +optional
     IPAMLifecycle NetworkIPAMLifecycle `json:"ipamLifecycle,omitempty"`
+    
+    // Control how much of the IP configuration will be managed by OVN.
+    // `Full` means OVN-Kubernetes will apply IP configuration to the SDN infrastructure and it will also assign IPs from the selected subnet to the
+    // individual pods.
+    // `Disabled` means OVN-Kubernetes will only assign MAC addresses and provide layer 2 communication, letting users configure IP addresses for the pods. 
+    //`Disabled` is only available for `Layer2` and `Localnet` topologies for Secondary networks. 
+    // By disabling IPAM, any Kubernetes features that rely on selecting pods by IP will no longer function 
+    // (such as network policy, services, etc). Additionally, IP port security will also be disabled for interfaces attached to this network.
+    // Defaults to `Full`.
+    // +optional
+    IPAMMode IPAMMode `json:"ipamMode,omitempty"`
```

Comment:
I like the name of the attribute.

Could it be a *IPAMMode instead ?... 

```golang
type IPAMMode struct {}
```

and we'd then add more stuff to it if/once needed.

Reply:
Yeah I intend it to be an enum, I don't think we will add more stuff under IPAMMode, e.g. if we want to add a flag for portSecurity, it will likely be its own field, because it is not necessarily a part of IPAMMode (especially if we will put MAC-related options in there), or maybe you have some other candidates that you think will make sense as a future part of this struct?

Reply:
Yes, for sure port security would go in a different knob.